crates/api_crud/src/post/create.rs

```diff
@@ -50,17 +50,21 @@ pub async fn create_post(
 ) -> Result<Json<PostResponse>, LemmyError> {
   let local_site = LocalSite::read(&mut context.pool()).await?;
 
+  honeypot_check(&data.honeypot)?;
+
   let slur_regex = local_site_to_slur_regex(&local_site);
   check_slurs(&data.name, &slur_regex)?;
+
   let body = process_markdown_opt(&data.body, &slur_regex, &context).await?;
-  honeypot_check(&data.honeypot)?;
+  let alt_text = process_markdown_opt(&data.alt_text, &slur_regex, &context).await?;
```

Comment:
It looks like Mastodon only supports plaintext, I dont think markdown makes sense here.

Reply:
Also it should probably throw an error if alt text is set but the url is not an image. `fetch_link_metadata_opt()` gives you the mime type. Maybe custom thumbnail should also be limited to certain mime types like `image/*` and `video/*`.

Reply:
I looked into this, but it could cause more errors than is worth it. For example, a post alt_text gets updated, which doesn't change the URL, so it skips fetching the metadata, leaving a blank mimetype, and so would throw an error. 

This could also act as a *short extra description*, even for non-image posts with a thumbnail.

Reply:
Maybe you can read the metadata of the existing post from the db when updating.

The alt text would normally be rendered in the `title` attribute of an image, where would you render it if there is no image?

Reply:
That's up to front ends I spose, but it could also be on the thumbnail of a link post. Or on a placeholder image for a body post.